Rename llvm::ThreadPool -> llvm::DefaultThreadPool (NFC)

joker-eph opened the pull request:
The base class llvm::ThreadPoolInterface will be renamed llvm::ThreadPool
in a subsequent commit.

This is a breaking change: clients who use to create a ThreadPool must now
create a DefaultThreadPool.

kuhar reviewed (approved):
Makes sense to me

joker-eph commented:
@dwblaikie : how would you split it? I didn't quite get the two renamings you have in mind?

dwblaikie commented:
> @dwblaikie : how would you split it? I didn't quite get the two renamings you have in mind?

One patch `ThreadPool->DefaultThreadPool` (people get a build error about `ThreadPool` not being the name of anything, find this patch as the root cause, and rename all their ThreadPool->DefaultThreadPool)
Follow-up patch, `ThreadPoolInterface->ThreadPool` (similarly clear/separate errors)

Changing both in one patch risks folks getting confusing error messages because their existing ThreadPool usage will now instantly start being interpreted as the interface type - resulting in different/confusing errors about inability to instantiate abstract types, etc, presumably. Rather than just that the name is no longer present at all.

Ultimately they can root cause and figure out both renamings - probably not a huge deal either way, but my understanding was that separating them might be marginally better for downstrteamers.

pogo59 commented:
> separating them might be marginally better for downstrteamers.

Hear, hear.

joker-eph commented:
> One patch ThreadPool->DefaultThreadPool (people get a build error about ThreadPool not being the name of anything, find this patch as the root cause, and rename all their ThreadPool->DefaultThreadPool)

Gotcha, thanks for elaborating, somehow my brain was slow on a Monday.
Will look into this!

joker-eph commented:
I did the first part of the renaming @dwblaikie : looks good?

dwblaikie commented:
> I did the first part of the renaming @dwblaikie : looks good?

Hmm - this patch still seems to have both renamings in it, if I'm reading the PR correctly? I take it from the subject that isn't the intent/the intent is that his patch only does the ThreadPool->DefaultThreadPool change?

joker-eph commented:
Actually no: the first patch landed already, so `ThreadPoolInterface` is now the base class in the codebase. I have some mixup here in that when renaming ThreadPool -> DefaultThreadPool, I used the base class ThreadPoolInterface when updating some of the uses.

I will push these ahead as a NFC change and rebase, that'll reduce the diff here.

joker-eph committed:
Rename llvm::ThreadPool -> llvm::DefaultThreadPool (NFC)

The base class llvm::ThreadPoolInterface will be renamed llvm::ThreadPool
in a subsequent commit.

joker-eph commented on lldb/source/Core/Debugger.cpp:
```diff
@@ -104,7 +104,7 @@ static std::recursive_mutex *g_debugger_list_mutex_ptr =
 static Debugger::DebuggerList *g_debugger_list_ptr =
     nullptr; // NOTE: intentional leak to avoid issues with C++ destructor chain
-static llvm::ThreadPool *g_thread_pool = nullptr;
+static llvm::DefaultThreadPoolThreadPool *g_thread_pool = nullptr;
```
Weirdly this issue wasn't caught in CI!
